### PR TITLE
GSdx: Folder Reorganization

### PR DESCRIPTION
**Pull request by arcum42**
This pull request is for the pending reorganization of the folder structure on GSdx, making it better organized and easier to work with.

This does not have to be merged immediately, and might be better merged after any other pending GSdx changed, so it can be based on them. I also thought I'd give a chance for any suggestions for files to be moved or additional folders to be created before it gets merged.

Renaming the files as well is still a possibility, but I think it might be better to just do the merge and folder creation at the moment and minimize confusion.

### CODE REVIEW

**Comment by lightningterror** [comment]
I prefer if we have proper commit naming starting with **GSdx:**, it's pretty much a naming standard at this point. Same goes for other type of changes (plugins, core, db).

**Comment by lightningterror** [comment]
Last I checked windows worked so it should be ok ? Needs a rebase now tho, also I prefer to merge this after https://github.com/PCSX2/pcsx2/pull/2344

**Comment by mirh** [comment]
Could also wait for TAS pr imho...
It's already yuge and tiresome enough I wouldn't want even the slightest of avoidable problems to hit it

**Comment by lightningterror** [comment]
I forgot about that one.

**Comment by willkuer** [comment]
TAS isn’t affecting gsdx, is it?

**Comment by mirh** [comment]
It does [a little](https://github.com/PCSX2/pcsx2/pull/2412/commits/494a98392bec57b4819b8197ef70675e2f1ceb0a), for recording
Iirc that may get decoupled, for another time, but I wouldn't want to cause unneeded complications

**Comment by lightningterror** [comment]
We can merge this now since TAS needs updating anyway.

**Comment by arcum42** [comment]
It's a busy monday and I've got a few hours to go on work, but my last rebase was after TC. And GSDx was working at that point. 

You did one commit since then, but that didn't touch any of the files involved. We're probably fine on merging it.

**Comment by lightningterror** [comment]
What about the old boost_spsc_queue.hpp file ? Shouldn't it be removed ?

**Comment by arcum42** [comment]
Right, I did forget to go back and delete that, didn't I? Taken care of. Not tested in Windows, though, so we'll see if it compiles there.

File: plugins/GSdx/CMakeLists.txt [hunk 1]
```diff
@@ -198,64 +159,103 @@ set(GSdxHeaders
...
+        Renderers/DX11/GSRendererDX11.cpp
+        Renderers/DX9/GSRendererDX9.cpp
+        Renderers/DXCommon/GSRendererDX.cpp
+        Renderers/DX11/GSTexture11.cpp
+        Renderers/DX9/GSTexture9.cpp
+        Renderers/DX11/GSTextureCache11.cpp
+        Renderers/DX9/GSTextureCache9.cpp
+        Renderers/DX11/GSTextureFX11.cpp
+        Renderers/DX9/GSTextureFX9.cpp
+        Renderers/DXCommon/GSTextureFX.cpp
```
Comment (turtleli) [inline plugins/GSdx/CMakeLists.txt]: This file seems to be unused.

**Comment by turtleli** [comment]
I don't see how that affects the placeholder being unnecessary?

**Comment by arcum42** [comment]
I don't see any reason to delete it right now.

File: plugins/GSdx/CMakeLists.txt (hunk 1 above)
Comment (arcum42) [reply plugins/GSdx/CMakeLists.txt]: Yes. The cmake files do include the Windows files, even though they are not used, and I updated them to reflect the new paths.

Comment (lightningterror) [reply plugins/GSdx/CMakeLists.txt]: The file GSTextureFX.cpp isn't included in the VS build files so it's only on CMakeList. I believe it can be purged.

**Comment by lightningterror** [comment]
Should the Scanline files be in SW render folder ?

**Comment by lightningterror** [comment]
Let's merge it then.

**Pull request merged**